src/jit/compiler.cpp

```diff
@@ -11097,3 +11097,831 @@ bool Compiler::killGCRefs(GenTree* tree)
+    m_VariableLiveRanges->back().m_StartEmitLocation.CaptureLocation(_emitter);
+
+    // startEmitLocationendEmitLocation has to be Valid and endEmitLocationendEmitLocation  not
+    noway_assert(m_VariableLiveRanges->back().m_StartEmitLocation.Valid());
+    noway_assert(!m_VariableLiveRanges->back().m_EndEmitLocation.Valid());
+}
+
+#ifdef DEBUG
+void VariableLiveDescriptor::dumpAllRegisterLiveRangesForBlock(emitter* _emitter, const CodeGenInterface* codeGen) const
+{
+    // "m_VariableLiveRanges" should has been initialized
+    noway_assert(m_VariableLiveRanges != nullptr);
+
+    if (m_VariableLiveRanges->empty())
+    {
+        printf("None history\n");
+    }
+    else
+    {
+        printf("[");
+        for (LiveRangeListIterator it = m_VariableLiveRanges->begin(); it != m_VariableLiveRanges->end(); it++)
+        {
+            dumpVariableLiveRange(&it, _emitter, codeGen);
+        }
+        printf("]\n");
+    }
+}
+
+void VariableLiveDescriptor::dumpRegisterLiveRangesForBlockBeforeCodeGenerated(const CodeGenInterface* codeGen) const
+{
+    // "m_VariableLifeBarrier" should has been initialized
+    noway_assert(m_VariableLifeBarrier != nullptr);
+
+    if (hasVarHomesToDump())
+    {
+        printf("[");
+        for (LiveRangeListIterator it = m_VariableLifeBarrier->getStartForDump(); it != m_VariableLiveRanges->end();
+             it++)
+        {
+            dumpVariableLiveRange(&it, codeGen);
+        }
+        printf("]\n");
+    }
+    else
+    {
+        printf("None history\n");
+    }
+}
+
+// Returns true if a live range for this variable has been recorded from last call to EndBlock
+bool VariableLiveDescriptor::hasVarHomesToDump() const
+{
+    // "m_VariableLifeBarrier" should has been initialized
+    noway_assert(m_VariableLifeBarrier != nullptr);
+
+    return m_VariableLifeBarrier->hasLiveRangesToDump();
+}
+
+// Reset the barrier so as to dump only next block changes on next block
+void VariableLiveDescriptor::endBlockLiveRanges()
+{
+    noway_assert(m_VariableLifeBarrier != nullptr);
+
+    // make "m_VariableLifeBarrier->m_StartingLiveRange" now points to nullptr for printing purposes
+    m_VariableLifeBarrier->resetDumper(m_VariableLiveRanges);
+}
+
+#endif // DEBUG
+
+//------------------------------------------------------------------------
+//                      VariableLiveKeeper
+//------------------------------------------------------------------------
+
+void Compiler::initializeVariableLiveKeeper()
+{
+    CompAllocator allocator = getAllocator(CMK_VariableLiveRanges);
+
+    int amountTrackedVariables = opts.compDbgInfo ? info.compLocalsCount : 0;
+    int amountTrackedArgs      = opts.compDbgInfo ? info.compArgsCount : 0;
+
+    varLiveKeeper = allocator.allocate<VariableLiveKeeper>(1);
+    memset(varLiveKeeper, 0, sizeof(*varLiveKeeper));
+    new (varLiveKeeper, jitstd::placement_t()) VariableLiveKeeper(amountTrackedVariables, amountTrackedArgs, this);
```

Comment:
`varLiveKeeper = new (allocator) VariableLiveKeeper(…`

Reply:
How could I check it does what I want?

Reply:
I'm not sure I understand the question. This is the normal way to allocate a new object in the JIT.

Reply:
Thanks, I didn't know that